src/OrbitHttp/HttpDownloadManagerTest.cpp

```diff
@@ -0,0 +1,210 @@
+
+namespace orbit_http {
+using orbit_base::CanceledOr;
+using orbit_base::FileExists;
+using orbit_base::Future;
+using orbit_base::IsCanceled;
+using orbit_base::StopSource;
+using orbit_base::TemporaryFile;
+using orbit_test_utils::HasError;
+using orbit_test_utils::HasNoError;
+
+namespace {
+class HttpDownloadManagerTest : public ::testing::Test {
+ protected:
+  HttpDownloadManagerTest()
+      : executor_(orbit_qt_utils::MainThreadExecutorImpl::Create()), manager_() {
+    local_http_server_process_ = new QProcess();
+#ifdef _WIN32
+    local_http_server_process_->setProgram("py");
+    local_http_server_process_->setArguments(
+        QStringList{"-3", "-m", R"(http.server)", "--bind", "localhost", "--directory",
+                    QString::fromStdString(orbit_test::GetTestdataDir().string()), "0"});
+#else
+    local_http_server_process_->setProgram("python3");
+    local_http_server_process_->setArguments(
+        QStringList{"-m", R"(http.server)", "--bind", "localhost", "--directory",
+                    QString::fromStdString(orbit_test::GetTestdataDir().string()), "0"});
+#endif
+
+    QProcessEnvironment current_env = local_http_server_process_->processEnvironment();
+    current_env.insert("PYTHONUNBUFFERED", "true");
+    local_http_server_process_->setProcessEnvironment(current_env);
+
+    ORBIT_LOG("Execute command:\n\"%s %s\"\n", local_http_server_process_->program().toStdString(),
+              local_http_server_process_->arguments().join(" ").toStdString());
+
+    QEventLoop loop{};
+    QObject::connect(local_http_server_process_, &QProcess::readyReadStandardOutput,
+                     [&loop, this]() {
+                       const QString prefix = "Serving HTTP on";
+                       QString std_output = local_http_server_process_->readAllStandardOutput();
+                       if (!std_output.contains(prefix)) return;
+
+                       QRegularExpression portRegex("port ([0-9]+)");
+                       QRegularExpressionMatch portMatch = portRegex.match(std_output);
+                       if (portMatch.hasMatch()) {
+                         port_ = portMatch.captured(1);
+                         loop.quit();
+                       }
+                     });
+
+    QObject::connect(local_http_server_process_, &QProcess::errorOccurred,
+                     [&loop, this](QProcess::ProcessError error) {
+                       if (error == QProcess::Crashed) return;
+                       ORBIT_LOG("Error while executing process.\nError:\n%s,\nDetails:\n%s.\n",
+                                 QMetaEnum::fromType<QProcess::ProcessError>().valueToKey(error),
+                                 local_http_server_process_->errorString().toStdString());
+                       if (loop.isRunning()) loop.quit();
+                     });
+
+    local_http_server_process_->start();
+    loop.exec();
+  }
+
+  ~HttpDownloadManagerTest() override {
+    local_http_server_process_->kill();
```

Comment:
Below I suggest to make QProcess a value which means it will be automatically deleted and will also automatically kill the process. So this line is not needed anymore.